src/js/prop-types/locale-prop-type.js

```diff
@@ -0,0 +1,13 @@
+import isRequiredHelper from './is-required-helper';
+
+// link https://stackoverflow.com/questions/8758340/regex-to-detect-locales
+const reLocale = /^[a-z]{2,4}(-([A-Z][a-z]{3}|[0-9]{3}))?(-([A-Z]{2}|[0-9]{3}))?$/;
```

Comment:
Well, maybe this is overkill and we just need checks for language and country code?